interfaces/dns_record/v0/README.md

```diff
@@ -0,0 +1,110 @@
+# `dns_record`
+
+## Usage
+
+This relation interface describes the expected behavior of any charm claiming to be able to interface with a DNS record Provider.
+
+## Direction
+
+The `dns_record` interface implements a provider/requirer pattern. The requirer is a charm that wishes to create a set of DNS records, and the provider is the charm managing those.
+```mermaid
+flowchart TD
+  Requirer -- dns-domains, dns-entries --> Provider
+  Provider -- dns-domains, dns-entries --> Requirer
+```
+
+## Behavior
+
+The following is the criteria that a Provider and Requirer need to adhere to be compatible with this interface.
+
+### Provider
+
+- Is expected to provide a list of dns-domains and a list of dns-entries in the relation databag, each containing the domain, the status and optionally the description corresponding to the dns-domains and dns-entries requirested by the requirer.
+- Is expected to authenticate requests for dns-domains based on internal business rules/processes at the organisation where this charm is deployed.
+
+### Requirer
+
+- Is expected to provide a list of dns-domains in the relation databag, each containing the domain it is requesting a DNS record for, the username and the password containing a juju secret for the provider to authenticate the request.
+- Is expected to provide a list of dns-entries mains in the relation databag, containing at least the dns-domain, the host-label and record-data. The dns-domain must be present in the list of dns-domains for authentication.
+
+
+## Relation Data
+
+### Provider
+
+[\[JSON Schema\]](./schemas/provider.json)
+
+Provider provides the result of the requirer request. It should be placed in the application databag.
+
+#### Example
+```json
+  "application-data": {
+    "dns-domains": [
+      {
+        "uuid": "550e8400-e29b-41d4-a716-446655440000",
+        "status": "invalid_credentials",
+        "description": "invalid_credentials"
+      },
+      {
+        "uuid": "550e8400-e29b-41d4-a716-446655440001",
+        "status": "approved"
+      }
+    ],
+    "dns-entries": [
+      {
+        "uuid": "550e8400-e29b-41d4-a716-446655440002",
+        "status": "invalid_credentials",
+        "description": "invalid_credentials"
+      },
+      {
+        "uuid": "550e8400-e29b-41d4-a716-446655440003",
+        "status": "approved"
+      }
+    ]
+  }
+
+```
+
+### Requirer
+
+[\[JSON Schema\]](./schemas/requirer.json)
+
+Requirer request the details of one or more DNS records. It should be placed in the application databag.
+
+#### Example
+
+```json
+  "application-data": {
+    "dns-domains": [
+      {
+        "uuid": "550e8400-e29b-41d4-a716-446655440000",
```

Comment:
Why the requirer requests those uuid's? When I go on AWS Route53 and add domains, I don't decide my uuid's, those are generated in the background.

Reply:
The idea is to map those to the provider responses

Reply:
Fwiw, the intention is that the relation library would generate these UUIDs for you, so it wouldn't be something end users would need to decide for themselves. As Arturo mentions it's just so that it's easier to track requests programmatically (since this is relation data, it's all being handled via the charms).